File: app/src/main/java/net/myacxy/squinch/helpers/tracking/builder/LogBuilder.java
package net.myacxy.squinch.helpers.tracking.builder;

import android.support.annotation.NonNull;

import net.myacxy.squinch.helpers.tracking.ITracker;
import net.myacxy.squinch.helpers.tracking.TrackingHelper;

public class LogBuilder extends PropertyLogBuilder<LogBuilder> {

    public LogBuilder(TrackingHelper trackingHelper) {
        super(trackingHelper);
    }

    @Override
    public TrackingHelper post() {
        for (ITracker tracker : trackers) {
            tracker.log(properties);
        }
        return trackingHelper;
    }

    @NonNull
    @Override
    protected LogBuilder self() {
        return this;
    }
}

File: app/src/main/java/net/myacxy/squinch/helpers/tracking/builder/LogBuilder.kt
package net.myacxy.squinch.helpers.tracking.builder

import net.myacxy.squinch.helpers.tracking.TrackingHelper

class LogBuilder(trackingHelper: TrackingHelper) : PropertyLogBuilder<LogBuilder>(trackingHelper) {

    override fun post(): TrackingHelper {
        trackers.forEach { it.log(booleans, ints, strings) }
        return trackingHelper
    }

    override fun self(): LogBuilder {
        return this
    }
}
